### PR TITLE
Ready for review. @AnnJohn @jmburges

### PR DESCRIPTION
**Pull request by queerviolet**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+You'll notice that the installer prints a big notice of several things you should do. In particular, we should have a route route.
+
+Create a `WelcomeController` with a `home` action and a view that just prints `current_user.email`.
+
+Now generate your `User` model with:
+
+    rails g devise User
+
+Run `rake routes`. You should see that Devise has added a bunch of them. Run `rails s` and take a look at one, maybe `/users/sign_in`.
+
```
Comment (adamjonas) [inline README.md]: There is a migration step missing here. The error would be obvious. Not sure how explicit we need to be. @jmburges?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+Now let's add Facebook login support with Omniauth.
+
+Add omniauth-facebook to your Gemfile:
+
+    gem 'omniauth-facebook'
+
+We'll also need to store two more columns in our user model: a `provider` string (which will always be `'facebook'` or `nil` in our app for the moment), and a `uid`, the user's authenticated Facebook ID.
+
+    rails g migration AddOmniauthToUsers provider:index uid:index
+    rake db:migrate
```
Comment (adamjonas) [inline README.md]: This one was explicit. See https://github.com/learn-co-curriculum/devise_lab/pull/1/files#r49518891

File: README.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+And add the Omniauth configuration to `config/initializers/devise.rb`:
+
+    config.omniauth :facebook, ENV['FACEBOOK_KEY'], ENV['FACEBOOK_SECRET']
```
Comment (adamjonas) [inline README.md]: There is a commented out line on 241 that shows a similar config for github, might want to reference it or instruct to put near by.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+This requires that you have `FACEBOOK_KEY` and `FACEBOOK_SECRET` in your environment. Create an application in the [Facebook developer console][fbdev] and get these values from there. You'll also need to go to Settings > Advanced and set Valid OAuth Redirect URLs to include `http://localhost:3000/users/auth/facebook/callback`.
```
Comment (adamjonas) [inline README.md]: The oauth section on FB could be a little more clear. There is a box above it that has space for server IP whitelist that I could see folks getting confused about. I'd call out the `Client OAuth Settings` section and then refer to the field.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+Add this line to your `user.rb` to enable omniauth:
+
+    devise :omniauthable, :omniauth_providers => [:facebook]
+
+And add this line to your welcome view:
+
+    <%= link_to "Sign in with Facebook", user_omniauth_authorize_path(:facebook) %>
+
+If you visit your page and click "Sign in with Facebook," you should be able to go through a Facebook application authorization flow.
```
Comment (adamjonas) [inline README.md]: Don't believe we ever added this route.

Comment (adamjonas) [inline README.md]: Also based on the instructions on line 23, I've got email on current_user which would be confusing.

File: README.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+You'll be greeted with an error: `The action 'facebook' could not be found for Devise::OmniauthCallbacksController`. Okay, let's make one.
+
+Add this to `routes.rb` to tell create a route for Omniauth to send its authentication data to:
+
+    devise_for :users, :controllers => { :omniauth_callbacks => "users/omniauth_callbacks" }
```
Comment (adamjonas) [inline README.md]: Cool. So make sure to clear out the `devise_for :users` line or you will get a routes conflict.

**Comment by adamjonas** [comment]
So when the whole thing works you get the little 
screen shot 2016-01-12 at 5 20 17 pm
thingy... that should definitely be in there as should a prompt for them to check there DB and see that a user was created with an email and uid and such... That that big payoff and should be celebrated!

File: README.md (hunk 6 above)
Comment (queerviolet) [inline README.md]: I'm confused about this. We're just adding this route for the first time; there shouldn't be any routes there yet.

Without this route, I was able to go through the Facebook auth flow and then get an error when I came back to the app. I want students to hit that error to drive home what routes get hit during the auth flow.

File: README.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+    rails g migration AddOmniauthToUsers provider:index uid:index
+    rake db:migrate
+
+And add the Omniauth configuration to `config/initializers/devise.rb`:
+
+    config.omniauth :facebook, ENV['FACEBOOK_KEY'], ENV['FACEBOOK_SECRET']
+
+You'll see that there's another similar line commented out for Github; feel free to reference it. This requires that you have `FACEBOOK_KEY` and `FACEBOOK_SECRET` in your environment. Create an application in the [Facebook developer console][fbdev] and get these values from there. You'll also need to go to Settings > Advanced and set Valid OAuth Redirect URLs to include `http://localhost:3000/users/auth/facebook/callback`. This setting is listed under `Client OAuth Settings` in the dashboard, and yes, I wish I could link you right to it.
+
```
Comment (blake41) [inline README.md]: i've seen students get really stuck on this stuff.  if we could record a little video, or do screenshots i think that would be enormously helpful.

File: README.md (hunk 2 above)
Comment (queerviolet) [inline README.md]: I made them both explicit.

File: README.md (hunk 3 above)
Comment (queerviolet) [inline README.md]: Added a comment to that effect.

File: README.md (hunk 5 above)
Comment (queerviolet) [inline README.md]: I added some instructions to add more necessary pieces to the view. But we want them to hit the error, then add the route (on line 74)

File: README.md (hunk 7 above)
Comment (queerviolet) [inline README.md]: I added a screenshot of my dev console.

**Pull request merged**